bugbug/models/performancebug.py

```diff
@@ -0,0 +1,133 @@
+            bug_features.HasURL(),
+            bug_features.HasW3CURL(),
+            bug_features.HasGithubURL(),
+            bug_features.Whiteboard(),
+            bug_features.BlockedBugsNumber(),
+            bug_features.Product(),
+            bug_features.HasRegressionRange(),
+            bug_features.BugReporter(),
+            bug_features.Priority(),
+            bug_features.HasCVEInAlias(),
+            bug_features.HasAttachment(),
+            bug_features.FiledVia(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.fileref(),
+            feature_cleanup.url(),
+            feature_cleanup.synonyms(),
+            feature_cleanup.hex(),
+            feature_cleanup.dll(),
+            feature_cleanup.crash(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(
+                        feature_extractors, cleanup_functions, rollback=True
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = ImblearnPipeline(
+            [
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.0001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.0001),
+                                "first_comment",
+                            ),
+                        ]
+                    ),
+                ),
+                ("sampler", BorderlineSMOTE(random_state=0)),
+                (
+                    "estimator",
+                    xgboost.XGBClassifier(n_jobs=utils.get_physical_cpu_count()),
+                ),
+            ]
+        )
+
+    def get_labels(self):
+        classes = {}
+
+        for bug_data in bugzilla.get_bugs():
+            bug_id = int(bug_data["id"])
+
+            if "cf_performance_impact" not in bug_data or bug_data[
+                "cf_performance_impact"
+            ] in ("?", "none"):
+                continue
+
+            classes[bug_id] = 1 if _is_performance_bug(bug_data) else 0
+
+        logger.info(
+            "%d performance bugs",
+            sum(label == 1 for label in classes.values()),
+        )
+        logger.info(
+            "%d non-performance bugs",
+            sum(label == 0 for label in classes.values()),
+        )
+
+        return classes, [0, 1]
+
+    def get_feature_names(self):
+        return self.clf.named_steps["union"].get_feature_names_out()
+
+    def overwrite_classes(self, bugs, classes, probabilities):
+        for i, bug in enumerate(bugs):
+            if bug["cf_performance_impact"] in ("low", "medium", "high"):
+                continue
```

Comment:
The `overwrite_classes` have no impact on the accuracy in the training stage. There is no point in excluding this.